src/App.vue

```diff
@@ -1,16 +1,32 @@
 <script setup lang="ts">
 import { createMongoAbility } from '@casl/ability'
 import { useSettings } from '@stores/useSettings'
-import { updateCredential } from './api'
+import { instantiateApiClient } from '@services/ApiService'
 import { useAbility } from '@casl/vue'
 import LogService from '@services/LogService'
+import { useJwt } from '@vueuse/integrations/useJwt'
+import { useCookies } from '@vueuse/integrations/useCookies'
+import type { AuthPermission, AuthResource } from 'ccat-api'
+import type { JwtPayload } from 'jwt-decode'
 
-const { cookie, jwt } = storeToRefs(useSettings())
 const perms = useAbility()
 
+type AuthToken = JwtPayload & {
+	username: string
+	permissions: Record<AuthResource, AuthPermission[]>
+}
+
 onBeforeMount(() => {
+
+	const cookies = useCookies(['ccat_user_token'], { doNotParse: true, autoUpdateDependencies: true })
```

Comment:
you should move these computed variables and composables outside of the `onBeforeMount` hook, it should work anyway

Reply:
also i suggest to leave these things in the settings store as it was before, it's cleaner imho

Reply:
I moved everything (cookie, jwt and permissions) inside `useSettings`, renaming it `useMainStore`.
It was confusing because for every view there is a dedicated store, but I realized `useSettings` was not the store for `SettingsView.vue` XD